### PR TITLE
Add docker-compose E2E tests workflow for CI

### LINKED ISSUE #174: Testing: Add docker-compose E2E test in CI
## Summary

Add end-to-end testing of the full docker-compose stack (Caddy + FastAPI + SQLite) in CI.

## Acceptance Criteria

- [ ] CI workflow starts docker-compose in test mode
- [ ] Full user workflow tested:
  - Push artifact
  - Tag artifact
  - Get artifact via API
  - Download via direct Caddy route
- [ ] Verifies Caddy routing works correctly
- [ ] Tests timeout/retry behavior

## Implementation Notes

Could use GitHub Actions service containers or docker-compose directly in the workflow.

```yaml
# Example workflow step
- name: Start services
  run: docker compose up -d --build
  
- name: Wait for healthy
  run: |
    timeout 60 bash -c 'until curl -s http://localhost:8080/health; do sleep 1; done'
    
- name: Run E2E tests
  run: uv run pytest tests/e2e/ --e2e-url=http://localhost:8080
```

## Reference

From proposed-issues.md Issue M.

---
(AI-generated via Claude Code)

### PR DESCRIPTION
**Pull request by duplico**
## Summary

- Add new GitHub Actions workflow `.github/workflows/e2e.yml` that runs E2E tests against the full docker-compose stack
- Workflow builds Docker images, starts services, and runs the `just e2e` recipe
- Includes failure diagnostics with docker-compose logs on test failure

## Test Plan

- [ ] Verify the E2E workflow runs successfully in CI
- [ ] Confirm docker-compose build completes
- [ ] Confirm all E2E tests pass (core workflow, auth, edge cases, privilege dropping, observability)
- [ ] Verify logs are captured on failure for debugging

Closes #174

---
(AI-generated via Claude Code w/ Opus 4.5)

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          python-version: "3.13"
+
+      - name: Set up just
+        uses: extractions/setup-just@v2
+
+      - name: Install dependencies
+        run: uv sync --all-groups
+
+      - name: Build Docker image
+        run: docker compose build
```
[the remark this answers is not in the document]
Comment (duplico) [reply .github/workflows/e2e.yml:49]: The pre-build step is intentional for CI observability - it separates build failures from test failures in the workflow logs and ensures the image is cached. The fixture will use the cached image. MAGPIE_DATA_DIR is a runtime concern, not relevant to the build step.

(AI-generated via Claude Code w/ Opus 4.5)

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+      - name: Run E2E tests
+        run: just e2e
+
+      - name: Show docker-compose logs on failure
+        if: failure()
+        run: docker compose logs --tail=200
```
[the remark this answers is not in the document]
Comment (duplico) [reply .github/workflows/e2e.yml]: Fixed in 0685b04. Removed the step since containers are cleaned up by the pytest fixture's finally block before this would run.

(AI-generated via Claude Code w/ Opus 4.5)

File: tests/e2e/test_privilege_dropping.py [hunk 3]
```diff
@@ -225,142 +264,146 @@ def test_container_rejects_non_numeric_gid(self) -> None:
...
-                    "-c",
-                    "cat /run/magpie-user && stat -c '%a' /run/magpie-user",
-                ],
-                capture_output=True,
-                text=True,
-                check=True,
-            )
+                actual_uid = int(numeric_lines[-2]) if len(numeric_lines) >= 2 else int(lines[-3])
+                actual_gid = int(numeric_lines[-1]) if len(numeric_lines) >= 1 else int(lines[-2])
+                username = text_lines[-1] if text_lines else lines[-1]
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/e2e/test_privilege_dropping.py:314]: Valid feedback - the parsing logic in these tests could be simplified to use the _extract_id_output helper consistently. Creating a follow-up issue to track this refactoring work.

(AI-generated via Claude Code w/ Opus 4.5)
